tcp_check/datadog_checks/tcp_check/tcp_check.py

```diff
@@ -95,6 +104,22 @@ def should_resolve_ips(self):
             return False
         return get_precise_time() - self.ip_cache_last_ts > self.ip_cache_duration
 
+    def has_ipv6_connectivity(self):
+        # type: () -> bool
+        try:
+            for _, _, _, _, sockaddr in socket.getaddrinfo(
+                socket.gethostname(), None, socket.AF_INET6, 0, socket.IPPROTO_TCP
+            ):
+                if not sockaddr[0].startswith('fe80:'):
+                    self.log.debug('Host %s is reporting IPv6 connectivity.', socket.gethostname())
+                    return True
+            return False
+        except socket.gaierror as e:
+            self.log.warning(
+                "Encountered error checking host's IPv6 connectivity with hostname %s: %s", socket.gethostname(), str(e)
+            )
+            return False
```

Comment:
Not sure if this is a reliable way to check for IPv6 connectivity. If the hostname returned from `socket.gethostname()` does not map to an IPv6 address (i.e. in /etc/hosts), then `socket.getaddrinfo(socket.gethostname(), None, socket.AF_INET6)` will return `[Errno 8] nodename nor servname provided, or not known` and has_ipv6_connectivity() will return False. 

In local testing on MacOS with confirmed IPv6 connectivity, this method returns `False` because the machine's hostname is not mapped to any IP addresses.